addon/services/alexandria-documents.js

```diff
@@ -131,6 +135,64 @@ export default class AlexandriaDocumentsService extends Service {
     await fileModel.save();
   }
 
+  /**
+   * Moves one or multiple files to a new category.
+   *
+   * @param {Object} newCategory category instance.
+   * @param {Array<Number>} documentIds.
+   */
+  async move(newCategory, documentIds) {
+    const INVALID_FILE_TYPE = "invalid-file-type";
+
+    const states = await Promise.all(
+      documentIds.map(async (id) => {
+        const document = this.store.peekRecord("document", id);
+
+        if (!document || document.category.id === newCategory) {
+          return true;
+        }
+
+        if (!fileHasValidMimeType(document, newCategory)) {
+          return "invalid-file-type";
+        }
+
+        const previousCategory = this.store.peekRecord(
+          "category",
+          document.category.id,
+        );
+
+        try {
+          document.category = newCategory;
+          await document.save();
+          return true;
+        } catch (error) {
+          document.category = previousCategory;
+
+          new ErrorHandler(this, error).notify();
+
+          return false;
+        }
+      }),
+    );
+
+    if (states.includes(INVALID_FILE_TYPE)) {
+      this.notification.danger(
+        this.intl.t("alexandria.errors.invalid-file-type", {
+          category: newCategory.name,
+          types: newCategory.allowedMimeTypes
+            .map((t) => mime.getExtension(t))
+            .filter(Boolean)
+            .join(", "),
+        }),
+      );
```

Comment:
Do you want to extract this into a method `mimeTypeErrorNotifcation`?

Reply:
Yep, i'll do that.